### PR TITLE
Update the Help Embed

### PR DESCRIPTION
**Pull request by GlenMerlin**
### **Please describe all changes made by this Pull Request and why you feel like it should be merged:**
Improved the help embed and added categories for each command (the code to determine which one is going to be used to generate the help embed is probably poorly optimized but its what works right now and I can fix it later or in another PR if you want

### **Checklists**
_Please tick the following checklists (by placing an x) as appropriate._

-   [x] I have read and agreed to both the [Code of Conduct](./CODE_OF_CONDUCT.md) and [Contributing Guidelines](.github/CONTRIBUTING.md) (**required**)
-   [x] Code changes have been tested **with a bot account on Discord**, or there are none. (**required**)
-   [x] Any code changes have been checked against ESLint (`npx eslint .`) and the TypeScript compiler (`npx tsc`) and there are no errors, or there are no code changes. (**required**)

### CODE REVIEW

**Comment by GlenMerlin** [comment]
is the typescript compile failing supposed to happen?

**Comment by Costpap** [comment]
It is not, I'll review the Pull Request and request for changes to be made where apropriate.

File: src/botconfig-example.example [hunk 1]
```diff
@@ -93,4 +93,4 @@ const loggingInfo = {
     } as LogChannel,
 };
 // Destructing, allows for easier access to properties.
-export const { corelog, guildlog, errorlog } = loggingInfo;
+export const { corelog, guildlog, errorlog } = loggingInfo;
```
Comment (Costpap) [inline src/botconfig-example.example:96]: Please revert this change (look at suggested change for reference) and change the file name back to `botconfig-example.ts` as I feel like this is not required.

```suggestion
export const { corelog, guildlog, errorlog } = loggingInfo;

```

**Comment by GlenMerlin** [comment]
that's really strange cause it was building fine on my pc

File: src/commands/avatar.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { parseUserMention } from '../utils/parse';
 
 export default {
     name: 'avatar',
+    level: 'user',
```
Comment (Costpap) [inline src/commands/avatar.ts:6]: To be honest, I don't feel like "levels" are needed.

File: src/commands/ban.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { parseUserMention } from '../utils/parse';
 
 export default {
     name: 'ban',
+    level: 'admin',
```
Comment (Costpap) [inline src/commands/ban.ts:6]: Adding on from my previous comment, I feel like that in this case, the `adminOnly` boolean can be used instead of `level: admin`.

File: src/commands/deploy.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { Message, Client } from 'discord.js';
 
 export default {
     name: 'deploy',
+    level: 'owner',
```
Comment (Costpap) [inline src/commands/deploy.ts:7]: Further adding on from my 2 previous comments, I think that in this case, the `ownerOnly` boolean can be used instead of `level: owner`.

File: src/commands/help.ts [hunk 5]
```diff
@@ -34,6 +40,52 @@ export default {
...
+        if (name === 'owner') {
+            const embed = new Discord.MessageEmbed()
+                .setColor(0x6293f5)
+                .setAuthor(`${client.user.username} Help`, client.user.displayAvatarURL({ format: 'png' }))
+                .setTitle("Here's a list of all my owner exclusive commands:")
+                .setTimestamp()
+                .setFooter(`You can send "${prefix}help [command name]" to get info on a specific command!`);
+
+            client.commands.forEach((command) => {
+                if (command.level === 'owner') {
```
Comment (Costpap) [inline src/commands/help.ts:75]: In order to avoid using levels, which I described in my above comments, this suggestion could be implemented.

```suggestion
                if (command.ownerOnly) {
```

File: src/commands/help.ts [hunk 6]
```diff
@@ -34,6 +40,52 @@ export default {
...
+        if (name === 'admin') {
+            const embed = new Discord.MessageEmbed()
+                .setColor(0x6293f5)
+                .setAuthor(`${client.user.username} Help`, client.user.displayAvatarURL({ format: 'png' }))
+                .setTitle("Here's a list of all my commands:")
+                .setTimestamp()
+                .setFooter(`You can send "${prefix}help [command name]" to get info on a specific command!`);
+
+            client.commands.forEach((command) => {
+                if (command.level === 'admin') {
```
Comment (Costpap) [inline src/commands/help.ts:52]: In order to avoid using levels, which I described in my above comments, this suggestion could be implemented:

```suggestion
                if (command.adminOnly) {
```

**Review by Costpap — CHANGES_REQUESTED** [review]
This pull request needs some changes, which I have mentioned below. The TypeScript compiler error needs to be fixed as well, but otherwise it looks fine to me.

File: src/typings/index.d.ts [hunk 7]
```diff
@@ -3,6 +3,8 @@ import { Message, Client, PermissionResolvable } from 'discord.js';
 export interface Command {
     /** The name of the command, should be identical to the file name. */
     name: string;
+    /** The Permission levels of this command */
```
Comment (Costpap) [inline src/typings/index.d.ts:6]: Assuming the removal of levels, this can be safely removed.

File: src/typings/index.d.ts [hunk 8]
```diff
@@ -3,6 +3,8 @@ import { Message, Client, PermissionResolvable } from 'discord.js';
...
+    level: string;
```
Comment (Costpap) [inline src/typings/index.d.ts:7]: Assuming the removal of levels, this can be safely removed.

```suggestion

```

**Review by Costpap — CHANGES_REQUESTED** [review]
Adding on from my previous review, please look at the comment above.

File: src/commands/help.ts [hunk 9]
```diff
@@ -21,7 +20,7 @@ export default {
                 .setFooter(`You can send "${prefix}help [command name]" to get info on a specific command!`);
 
             client.commands.forEach((command) => {
-                if (command.level === 'user') {
+                if (!command.adminOnly && !command.ownerOnly) {
```
Comment (Costpap) [inline src/commands/help.ts:23]: Shouldn't this be `||` instead of `&&`?

```suggestion
                if (!command.adminOnly || !command.ownerOnly) {
```

**Review by Costpap — CHANGES_REQUESTED** [review]
Everything looks fine, the only reason that changes are requested is because of a concern which is expressed in the comment below. Once that is addressed the pull request will be approved & merged.

File: src/commands/help.ts (hunk 9 above)
Comment (GlenMerlin) [reply src/commands/help.ts:23]: No | | makes the bot print the entire list of commands because one of the attributes is true and the other is false
&& prints just the non admin/owner only commands

**Review by Costpap — APPROVED** [review]
